Add the equivalent of 'kubernetes.io/ingress.allow-http: false' to IngressRoute

alexbrand opened the pull request:
Fixes #455 

Adds a new field to the IngressRoute to control whether HTTP is allowed on a given VirtualHost:

```
spec:
  virtualhost:
    fqdn: "example.com"
    allowHttp: false
```

Update (Aug 10, 2018): renamed the `allowHttp` field to `httpAllowed`:

```
spec:
  virtualhost:
    fqdn: "example.com"
    httpAllowed: false
```

davecheney commented on apis/contour/v1beta1/ingressroute.go:
```diff
@@ -39,6 +39,8 @@ type VirtualHost struct {
 	// matching certificate
 	TLS *TLS `json:"tls"`
+	// If set to false, this virtual host will only be accessible via HTTPS.
+	AllowHTTP *bool `json:"allowHttp"`
```
Let's use the past tense here, s/AllowHTTP/HTTPAllowed/g here and everywhere to be consistent.

alexbrand commented:
Thanks @davecheney. Renamed the field to `HTTPAllowed`:

```
spec:
  virtualhost:
    fqdn: "example.com"
    httpAllowed: false
```

Also added user-facing documentation.

alexbrand commented:
@davecheney Rebased and adapted implementation to the post-refactor dag builder. PTAL.
